app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -173,6 +173,9 @@ def quotas
   end
 
   def miq_templates
+
+    @vm_or_template_hash ||= database_query
```

Comment:
Is there a reason why this is being made an instance variable instead of local?

Reply:
It would be more proper way to use an instance variable because of the multiple using and in order to have an access to it from different places.

Reply:
Ah, I see that you are caching the result so that you aren't repeating the query in miq_templates and vms. I think I would recommend putting this in the collector and using a similar caching pattern to the other collector methods, but this is fine.